jdaviz/configs/imviz/plugins/viewers.py

```diff
@@ -280,15 +283,52 @@ def get_link_type(self, data_label):
         if len(self.session.application.data_collection) == 0:
             raise ValueError('No reference data for link look-up')
 
-        # TODO: Brett Morris might want to look at this for
-        # https://github.com/spacetelescope/jdaviz/pull/2179
-        #     ref_label = self.state.reference_data ???
-        #
-        # The original links were created against data_collection[0], not necessarily
-        # against the current viewer reference_data
-        ref_label = self.session.application.data_collection[0].label
-
-        return self.jdaviz_helper.get_link_type(ref_label, data_label)
+        ref_label = self.state.reference_data.label
+        if data_label == ref_label:
+            return 'self'
+
+        if ref_label in self.state.wcs_only_layers:
+            return 'wcs'
+
+        link_type = None
+        for elink in self.session.application.data_collection.external_links:
+            elink_labels = (elink.data1.label, elink.data2.label)
+            if data_label in elink_labels and ref_label in elink_labels:
+                if isinstance(elink, LinkSame):  # Assumes WCS link never uses LinkSame
+                    link_type = 'pixels'
+                else:  # If not pixels, must be WCS
+                    link_type = 'wcs'
+                break  # Might have duplicate, just grab first match
+
+        if link_type is None:
+            raise ValueError(f'{data_label} not found in data collection external links')
+
+        return link_type
+
+    def _get_fov(self, data):
+        # compute the mean of the height and width of the
+        # viewer's FOV on ``data`` in world units:
+        x_corners = [
+            self.state.x_min,
+            self.state.x_max,
+            self.state.x_min,
+            self.state.x_max
+        ]
+        y_corners = [
+            self.state.y_min,
+            self.state.y_min,
+            self.state.y_max,
+            self.state.y_max
+        ]
+
+        y_corners, x_corners = self._get_real_xy(
```

Comment:
@bmorris3 , is there a reason why you swap x/y here or is this a typo? `_get_real_xy` returns x and then y, not the other way.